lib/redfish_api.py

```diff
@@ -260,15 +263,27 @@ def get_openapi_from_redfish_org(self):
         Method to get openapi.yaml schema file from openapi url of redfish organisation
         :return: Bool,string output for openapi schema file
         """
+
         try:
-            result = requests.get(self.config_dict["openapi_url"])
-            if result.status_code != 200:
-                raise Exception("Failed")
-            try:
-                self.openapi_dict = yaml.load(result.text, Loader=yaml.FullLoader)
-            except Exception as e:
-                logger.error("error msg: {}".format(e))
-                sys.exit(1)
+            #if len(self.openapi_dict) > 0:
+            #    pass
+            #    #return
+            if os.path.exists("result.yml") != True:
+                print("Not exists", self.config_dict["openapi_url"])
+                result = requests.get(self.config_dict["openapi_url"])
+                if result.status_code == 200:
+                    #raise Exception("Failed")
+                    try:
+                        self.openapi_dict = yaml.load(result.text, Loader=yaml.FullLoader)
+                        f = open("result.yml", "w")
+                        f.write(result.text)
+                        f.close()
+                    except Exception as e:
+                        logger.error("error msg: {}".format(e))
+                        sys.exit(1)
+            else:
+                with open('result.yml', 'r') as file:
```

Comment:
Instead of providing hardcoded filename, can keep filename = "result.yml", and then pass this filename variable at line number 271 and 285.

Reply:
addressed on the latest commit